doc/api/https.md

```diff
@@ -29,6 +29,10 @@ An [`Agent`][] object for HTTPS similar to [`http.Agent`][]. See
   Can have the same fields as for [`http.Agent(options)`][], and
   * `maxCachedSessions` {number} maximum number of TLS cached sessions.
     Use `0` to disable TLS session caching. **Default:** `100`.
+  * `servername` {string | boolean} the value of
```

Comment:
Boolean might be slightly misleading? It's really `{string | false}` since setting it to `true` won't do what one might expect...

Reply:
Beware that `false` will be the only type case here and will baffle doc tools ([`tools/doc/type-parser.js`](https://github.com/nodejs/node/blob/master/tools/doc/type-parser.js) will throw).

Reply:
That was exactly my reasoning. We might want to use `null` as @bnoordhuis has suggested to fix this!